cmd/frontend/graphqlbackend/search_results.go

```diff
@@ -2002,6 +2004,39 @@ func compareSearchResults(left, right SearchResultResolver, exactFilePatterns ma
 	return arepo < brepo
 }
 
+func (r *searchResolver) selectResults(results []SearchResultResolver) []SearchResultResolver {
+	value, _ := r.Query.StringValue(query.FieldSelect)
+	if value == "" {
+		return results
+	}
+	sm, _ := filter.SelectPathFromString(value) // error checked during query parsing
```

Comment:
```suggestion
	sm, _ := filter.SelectPathFromString(value) // Invariant: select is validated.
```

This is how I (personally) annotate these, makes it [easy to search for](https://sourcegraph.com/search?q=repo:%5Egithub%5C.com/sourcegraph/sourcegraph%24+invariant:&patternType=literal). Won't impose, this is more of an FYI.

Reply:
Side note: whenever this pattern crops up, it usually means that we should be putting the validated value in the internal data structure and not inspecting the query value directly. But we have a bunch of places in our code where we look at the query value, and doing that is not so simple in the current state (refactoring that is part of a larger effort). So don't change anything, this is great. Just adding some context.

Reply:
Awesome -- I had a question above specifically about this. Thanks!